### PR TITLE
op-mode: T6601: move "reset ip(v6) arp|neighbor" to "clear ip(v6) arp|neighbor"

### PR DESCRIPTION
**Pull request by c-po**
## Change Summary

Move `reset ip(v6) arp|neighbor` to `clear ip(v6) arp|neighbor`

In addition create a build-time symbolic link for:
  `show ip arp` -> `show ip neighbors`

This change also introduces new-style op-mode for IPv6 ND operations rather then directly calling iproute2

Parts should be manually backported to 1.4 sagitta to also provide the new CLI commands but keep the old ones for CLI stability reasons!

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [x] Other (please describe): Op-mode CLI change

## Related Task(s)

* https://vyos.dev/T6601

## Related PR(s)
* https://github.com/vyos/vyos-documentation/pull/1511

## How to test

```
cpo@LR2.wue3:~$ clear ip arp
Possible completions:
  address               Reset ARP cache for an IPv4 address
  interface             Reset ARP cache for interface
  table                 Flush entire ARP cache
```

```
cpo@LR2.wue3:~$ clear ipv6 neighbors
Possible completions:
  address               Reset ND cache for an IPv6 address
  interface             Reset IPv6 ND cache for interface
  table                 Flush entire IPv6 ND cache
```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly

### CODE REVIEW

**Comment by c-po** [comment]
@MergifyIo backport circinus

File: op-mode-definitions/clear-ip.xml.in [hunk 1]
```diff
@@ -7,6 +7,37 @@
...
         </properties>
         <children>
+          <node name="arp">
+            <properties>
+              <help>Reset Address Resolution Protocol (ARP) cache</help>
+            </properties>
+            <children>
+              <tagNode name="address">
+                <properties>
+                  <help>Reset ARP cache for an IPv4 address</help>
```
Comment (sever-sever) [inline op-mode-definitions/clear-ip.xml.in:17]: Should there also be a `clear ARP cache`?

File: op-mode-definitions/clear-ip.xml.in [hunk 2]
```diff
@@ -7,6 +7,37 @@
...
+                <command>sudo ${vyos_op_scripts_dir}/neighbor.py reset --family inet --address "$5"</command>
+              </tagNode>
+              <tagNode name="interface">
+                <properties>
+                  <help>Reset ARP cache for interface</help>
+                  <completionHelp>
+                    <script>${vyos_completion_dir}/list_interfaces</script>
+                  </completionHelp>
+                </properties>
+                <command>sudo ${vyos_op_scripts_dir}/neighbor.py reset --family inet --interface "$5"</command>
```
Comment (sever-sever) [inline op-mode-definitions/clear-ip.xml.in:31]: Should there be a `clear` option instead of `reset`

**Review by sever-sever — APPROVED** [review]

**Review by dmbaturin — COMMENTED** [review]
Top level op mode words have specific meanings in VyOS. `clear` is reserved for completely non-disruptive operations: clearing the screen, clearing rule counters... `reset` is for locally-disruptive operations.

Since removing a neighbor table entry can lead to lost packets or increased latencies for new connections, it doesn't qualify for `clear`, which is why it was called `reset`.

I'd prefer to keep that terminology consistent. BGP neighbors in Cisco IOS are also under `clear`, but we don't plan to mimic that, do we? ;)

**Pull request closed without merging**